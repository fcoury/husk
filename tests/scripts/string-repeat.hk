// Test repeat() method for strings
let s = "Hello";
let single_char = "A";
let empty = "";
let unicode = "cafÃ©";
let emoji = "ðŸ‘‹";

// Test basic repetition
println!(s.repeat(0));      // Should print: ""
println!(s.repeat(1));      // Should print: "Hello"
println!(s.repeat(3));      // Should print: "HelloHelloHello"

// Test single character repetition
println!(single_char.repeat(5));  // Should print: "AAAAA"
println!(single_char.repeat(0));  // Should print: ""

// Test empty string repetition
println!(empty.repeat(0));   // Should print: ""
println!(empty.repeat(5));   // Should print: ""

// Test Unicode repetition
println!(unicode.repeat(2)); // Should print: "cafÃ©cafÃ©"
println!(unicode.repeat(0)); // Should print: ""

// Test emoji repetition
println!(emoji.repeat(3));   // Should print: "ðŸ‘‹ðŸ‘‹ðŸ‘‹"

// Test with variables
let count = 2;
println!("test".repeat(count)); // Should print: "testtest"

// Test with expressions
println!("X".repeat(1 + 2));    // Should print: "XXX"

// Test chaining
println!("ab".repeat(2).repeat(2)); // Should print: "abababab"

// Edge case: Large repetition (but reasonable for test)
println!("x".repeat(10));       // Should print: "xxxxxxxxxx"