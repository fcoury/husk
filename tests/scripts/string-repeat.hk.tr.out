
Hello
HelloHelloHello
AAAAA



cafÃ©cafÃ©

ğŸ‘‹ğŸ‘‹ğŸ‘‹
testtest
XXX
abababab
xxxxxxxxxx
